descriptive-link-5effbb.md

```diff
@@ -0,0 +1,179 @@
+---
+id: 5effbb
+name: Link text is descriptive
+rule_type: atomic
+
+description: | 
+ This rule checks that all links describe their purpose
+
+accessibility_requirements: 
+  wcag20:2.4.4: # Link Purpose (In Context) 
+    forConformance: true
+    failed: not satisfied
+    passed: satisfied
+    inapplicable: further testing needed
+
+input_aspects:
+- DOM Tree
+- CSS Styling
+
+authors:
+- Carlos Duarte
+- Marie Trudelle 
+- Ramit Garg
+---
+
+## Applicability
+
+The rule applies to any HTML element with the [semantic role](#semantic-role) of `link` that has an [accessible-name](#accessible-name) that is not the empty string ("").
+
+## Expectation
+
+Both the [text content](#text-content), if [visible](#visible), and the [accessible-name](#accessible-name) of the target element, together with its [visible](#visible) [context](#context) describe the purpose of the link.
```

Comment:
I suggest not using a more general "context" definition, but creating a definition for specifically "programmatically determined link context" in line with the [WCAG definition for this](https://www.w3.org/TR/WCAG21/#dfn-programmatically-determined-link-context): 

> In HTML, information that is programmatically determinable from a link in English includes text that is in the same paragraph, list, or table cell as the link or in a table header cell that is associated with the table cell that contains the link.

This definition is in my experience narrower than the one I think many accessibility experts are using, which often also includes headings etc.

In the ACT-R definition we will probably in addition to the WCAG part of the definiton have to define whether "programmatically determinable link context" is things that are visible or things that are included in the accessibility tree. "Programatically determinable" indicates the latter, in my opinion.

This definition will also be crucial for making a 2.4.4 (A) version of the 2.4.9 (AAA) rule [Links with identical accessible names serve equivalent purpose](https://act-rules.github.io/rules/b20e66)

Reply:
@carlosapaduarte, there is a shot at a "programmatically determined link context" definition in this new rule proposal: https://github.com/act-rules/act-rules.github.io/pull/693/files#diff-242a35ac05c129a11753d39e952c744c
(that was all along supposed to come bundled with its 2.4.9 sibling rule, https://act-rules.github.io/rules/b20e66)

If you continue work on this rule, feel free to build on/edit/change the definiton for this rule. I will align with you/this rule, when I pick up further work on the other rule.